src/analytics/analyticsService.ts

```diff
@@ -1,76 +1,114 @@
 import * as os from 'os';
 import { Version } from '../common/version';
 import { GUAService } from './guaService';
-import { TelerikAnalyticsService } from './telerikAnalyticsService';
 import { AnalyticsBaseInfo, OperatingSystem } from './analyticsBaseInfo';
 import { Services } from '../services/extensionHostServices';
 import * as utils from '../common/utilities';
+import * as vscode from 'vscode';
+import * as uuid from "uuid";
 
 export class AnalyticsService {
+    private static HAS_ANALYTICS_PROMPT_SHOWN_KEY = "nativescript.hasAnalyticsPromptShown";
+    private static CLIENT_ID_KEY = "nativescript.analyticsClientId";
+    private static ANALYTICS_PROMPT_MESSAGE = "Help improve NativeScript Extension by allowing Progress to collect data usage. " +
```

Comment:
Help us improve the NativeScript extension by allowing Progress to collect anonymous usage data. For more information about the gathered information and how it is used, read our [privacy statement](https://www.telerik.com/about/privacy-policy). 

You can [disable the analytics and data collection](https://github.com/NativeScript/nativescript-vscode-extension/blob/master/README.md#how-to-disable-the-analytics) at any given time.

Do you want to enable analytics?